source/string.c

```diff
@@ -57,11 +65,25 @@ void aws_string_destroy_secure(struct aws_string *str) {
 }
 
 int aws_string_compare(const struct aws_string *a, const struct aws_string *b) {
+    AWS_PRECONDITION(aws_string_is_valid(a));
+    AWS_PRECONDITION(aws_string_is_valid(b));
+    if (a == b) {
+        return 0; /* strings identical */
+    }
+    if (a == NULL) {
```

Comment:
Similarly as above, is_valid precondition is too strict.

Reply:
So yeah, I'd change it to `!a || aws_string_is_valid(a)`